File: templates/Features/ShareTargetVB/ViewModels/SharedDataWebLinkViewModel.vb
Imports Windows.ApplicationModel.DataTransfer
Imports Windows.ApplicationModel.DataTransfer.ShareTarget

Imports Param_ItemNamespace.Helpers

Namespace ViewModels
    Public Class SharedDataWebLinkViewModel
        Inherits SharedDataViewModelBase

        Private _uri As Uri

        Public Property Uri As Uri
            Get
                Return _uri
            End Get
            Set
                [Param_Setter](_uri, value)
            End Set
        End Property

        Public Sub New()
        End Sub

        Public Overrides Async Function LoadDataAsync(shareOperation As ShareOperation) As Task
            Await MyBase.LoadDataAsync(shareOperation)

            PageTitle = "ShareTargetFeature_WebLinkTitle".GetLocalized()
            DataFormat = StandardDataFormats.WebLink
            Uri = Await shareOperation.GetWebLinkAsync()
        End Function
    End Class
End Namespace

File: code/TemplateStudioForUWP/Templates/Uwp/Features/ShareTarget.Prism/ViewModels/SharedDataWebLinkViewModel.cs
using System;
using System.Threading.Tasks;

using Windows.ApplicationModel.DataTransfer;
using Windows.ApplicationModel.DataTransfer.ShareTarget;

using Param_RootNamespace.Helpers;

namespace Param_RootNamespace.ViewModels
{
    public class SharedDataWebLinkViewModel : SharedDataViewModelBase
    {
        private Uri _uri;

        public Uri Uri
        {
            get => _uri;
            set => SetProperty(ref _uri, value);
        }

        public SharedDataWebLinkViewModel()
        {
        }

        public override async Task LoadDataAsync(ShareOperation shareOperation)
        {
            await base.LoadDataAsync(shareOperation);

            PageTitle = "ShareTargetFeature_WebLinkTitle".GetLocalized();
            DataFormat = StandardDataFormats.WebLink;
            Uri = await shareOperation.GetWebLinkAsync();
        }
    }
}
